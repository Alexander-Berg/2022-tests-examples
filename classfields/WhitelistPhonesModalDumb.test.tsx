import React from 'react';
import { shallow } from 'enzyme';

import type { PhoneError } from 'www-cabinet/react/dataDomain/settingsWhitelist/types';

import WhitelistPhonesModalDumb from './WhitelistPhonesModalDumb';

const whiteListModalProps = {
    isVisible: true,
    hideModal: () => {},
    postPhones: () => {},
    isAddPhonesButtonActive: true,
    quantityOfAddedPhones: 122,
    invalidPhoneNumbers: [
        { phone: 'восемь-девятсотшестнадцать-три-семерки-две-пятерки-сорок-три', error: 'PARSE' as PhoneError },
        { phone: '+791111122333', error: 'DUPLICATE' as PhoneError }, { phone: '+79167775543', error: 'DUPLICATE' as PhoneError },
    ],
    quantityOfNewPhoneNumbers: 0,
};

it('должен показать текст с проблемными телефонами', async() => {
    const tree = shallowRenderComponent();
    tree.setState({ showAdditionalInfo: true });

    expect(tree.find('.WhitelistPhonesModal__subtitle').at(1)).toMatchSnapshot();
});

function shallowRenderComponent(props = whiteListModalProps) {
    return shallow(
        <WhitelistPhonesModalDumb { ...props }/>,
    );
}
