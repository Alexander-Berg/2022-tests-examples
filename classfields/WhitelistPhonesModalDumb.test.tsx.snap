// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен показать текст с проблемными телефонами 1`] = `
<div
  className="WhitelistPhonesModal__subtitle"
>
  <div>
    Следующие номера телефонов уже были добавлены в
     
    личный
     
    кабинет:
     
    +791111122333, +79167775543
  </div>
  <div>
    Не удалось распознать:
     
    восемь-девятсотшестнадцать-три-семерки-две-пятерки-сорок-три
  </div>
</div>
`;
